Jenkinsfile

```diff
@@ -0,0 +1,187 @@
+def MACHINE = 'none'
+def machine = 'none'
+def HOME = 'none'
+
+pipeline {
+    agent { label 'built-in' }
+
+    options {
+        skipDefaultCheckout()
+    }
+
+    stages {
+
+        stage('Get Machine') {
+            agent { label 'built-in' }
+            steps {
+                script {
+                    MACHINE = 'none'
+                    for (label in pullRequest.labels) {
+                        echo "Label: ${label}"
+                        if ((label.matches("CI-Hera-Ready"))) {
+                            MACHINE = 'hera'
+                        } else if ((label.matches("CI-Orion-Ready"))) {
+                            MACHINE = 'orion'
+                        } else if ((label.matches("CI-Hercules-Ready"))) {
+                            MACHINE = 'hercules'
+                        }
+                    }
+                    machine = MACHINE[0].toUpperCase() + MACHINE.substring(1)
+                }
+            }
+        }
+
+        stage('Get Common Workspace') {
+            agent { label "${MACHINE}-emc" }
+            steps ( timeout(time: 1, unit: 'HOURS') ) {
+                script {
+                    properties([parameters([[$class: 'NodeParameterDefinition', allowedSlaves: ['built-in','Hera-EMC','Orion-EMC'], defaultSlaves: ['built-in'], name: '', nodeEligibility: [$class: 'AllNodeEligibility'], triggerIfResult: 'allCases']])])
+                    HOME = "${WORKSPACE}/TESTDIR"
+                    sh( script: "mkdir -p ${HOME}/RUNTESTS", returnStatus: true)
+                    pullRequest.addLabel("CI-${machine}-Building")
+                    if ( pullRequest.labels.any{ value -> value.matches("CI-${machine}-Ready") } ) {
+                        pullRequest.removeLabel("CI-${machine}-Ready")
```

Comment:
How will this work? `machine` is all upper-case e.g. `HERA` and the labels are `CI-Hera-Ready`?

Reply:
Yes that idea.  The controler had hera-emc and orion-emc as labels and GitHub had labels as Orion-Ready.